src/integrationTest/java/uk/gov/hmcts/reform/blobrouter/tasks/processors/BlobProcessor2Test.java

```diff
@@ -0,0 +1,75 @@
+package uk.gov.hmcts.reform.blobrouter.tasks.processors;
+
+import com.azure.storage.blob.BlobContainerClient;
+import com.azure.storage.blob.models.BlobItem;
+import com.azure.storage.blob.specialized.BlobLeaseClientBuilder;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.ActiveProfiles;
+import uk.gov.hmcts.reform.blobrouter.data.EnvelopeRepository;
+import uk.gov.hmcts.reform.blobrouter.services.BlobReadinessChecker;
+import uk.gov.hmcts.reform.blobrouter.services.storage.BlobContainerClientProvider;
+import uk.gov.hmcts.reform.blobrouter.services.storage.BlobDispatcher;
+import uk.gov.hmcts.reform.blobrouter.util.BlobStorageBaseTest;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.BDDMockito.given;
+import static uk.gov.hmcts.reform.blobrouter.data.model.Status.DISPATCHED;
+
+@ActiveProfiles("db-test")
+@SpringBootTest
+@ExtendWith(MockitoExtension.class)
+public class BlobProcessor2Test extends BlobStorageBaseTest {
```

Comment:
Putting '2' in the name because `BlobProcessor2Test` already exists and this is a complete rewrite.
I'll remove the original file (and remove this one) in next PR.